### ISSUE TITLE
ALign LoRA accuracy with HF

### ISSUE THREAD

**Issue by aolemila** [sgl-project/sglang-jax#617]
Align the LoRA accuracy with HF.

### `test_01_single_prompt_align_last_layer_logits_with_hf_on_cpu`

**Prompts**: Prompt1(LoRA).
**Number of Layers**: All layers.
**Environment**: TPU-V6e-1.
**Device**: Both of them use CPU.
**Comparison Object**: The logits of prefill and first decode token.
**Tolerance**: 1e-4.
**Methods**:

```python3
        tolerance = 1e-4
        np.testing.assert_allclose(
            hf_prefill_logits, sgl_prefill_logits, atol=tolerance, rtol=tolerance
        )
        np.testing.assert_allclose(
            hf_decode_logits, sgl_decode_logits, atol=tolerance, rtol=tolerance
        )
```

<img width="703" height="271" alt="image" src="https://github.com/user-attachments/assets/7bee7e44-4e82-46f3-8720-091d5e5d57ae" />

The following content are copied from [PR610](https://github.com/sgl-project/sglang-jax/pull/610).

### `test_02_multi_prompts_align_last_layer_logits_with_hf_on_tpu`

**Prompts**: Prompt1(LoRA), Prompt2(None), Prompt3(LoRA). Send them concurrently.
**Number of layers**: All layers.
**Environment**: TPU-V6e-1.
**Device**: HF uses CPU, SGLangJax uses TPU. Note: If SGLangJax used CPU, the results would be dramatically different. I do two experiments.
    - E1: When I use TPU for SGLangJax, the logits of first decode token are more reasonable. See the following figure.
    - E2: In order to check the wrong batch results are not brought by LoRA feature, I use the prompts without LoRAs and annotate the LoRA codes in engine, the results are the same.
    - In short, I uses TPU for SGLangJax to test the multi prompts.
**Comparison Object**: The logits of decode. Note: In SGLangJax, the different request may be prepared into different batches, and this makes the comparison is more difficult. So I use first decode token's logits.
**Tolerance**: 2e-9.
**Methods**:

```python3
            tolerance = 9e-2
            np.testing.assert_allclose(
                hf_decode_logits, sgl_decode_logits, atol=tolerance, rtol=tolerance
            )
```

<img width="1144" height="755" alt="image" src="https://github.com/user-attachments/assets/beed0b58-995b-44c3-9e17-4583268e4394" />
